StudentUEK-SwiftUI/Scenes/Schedule/ScheduleViewModel.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  ScheduleViewModel.swift
+//  StudentUEK-SwiftUI
+//
+//  Created by private on 11/06/2021.
+//
+
+import Combine
+import SwiftUI
+
+final class ScheduleViewModel: ObservableObject {
+    @Published var visibleSections: [ScheduleSection] = []
+    @Published var filterButtonTitle: String = "Filtry"
+    @Published var showFilters: Bool = false
+    @Published var showAddSubject = false
+    @Published var query: String = "" {
+        didSet {
+            withAnimation {
+                showFilters = false
+                applyFilters()
+            }
+        }
+    }
+    
+    @Published var subjectTypes: Set<SubjectType> = Set(SubjectType.allCases) {
+        didSet {
+            withAnimation { applyFilters() }
+            updateFiltersCount()
+        }
+    }
+    
+    private let scheduleLoader: ScheduleLocalStore
+    private let grouper: ScheduleSectionGroupingLogic
+    
+    private var fullSchedule: [ScheduleSection] = []
+    private var cancellableBag: Set<AnyCancellable> = []
+    
+    init(scheduleLoader: ScheduleLocalStore, grouper: ScheduleSectionGroupingLogic) {
+        self.scheduleLoader = scheduleLoader
+        self.grouper = grouper
+    }
+    
+    func loadData() {
+        ApiService<[Subject]>()
+            .loadSchedule(for: URL(string: "http://planzajec.uek.krakow.pl/index.php?typ=G&id=210171&okres=3&xml")!)
```

Comment:
Urls should be hidden under some layer. You shouldn't use them that way. 

You could create SheduleDataProvider where you hide this implementation details.